contracts/utils/ExitQueue.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+/*
+    ExitQueue.sol - fair-manager
+    Copyright (C) 2025-Present SKALE Labs
+    @author Dmytro Stebaiev
+
+    fair-manager is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    fair-manager is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with fair-manager.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+pragma solidity ^0.8.24;
+
+import { Fair, Timestamp } from "@skalenetwork/fair-manager-interfaces/units.sol";
+
+library ExitQueueLibrary{
+
+    struct ExitRequest {
+        Fair amount;
+        Timestamp releaseDate;
+    }
+
+    struct UserExitQueue {
+        uint256 lastId;
+        uint256 firstId;
+        mapping(uint256 id => ExitRequest request) requests;
+    }
+
+    struct ExitQueue {
+        mapping(address user => UserExitQueue userExitQueue) userQueues;
+        uint256 maxRequests;
+        Timestamp retrievingDelay;
+        Fair totalFair;
+    }
+
+    event RequestCreated(address indexed user, uint256 indexed id, Fair amount, Timestamp releaseDate);
+    event RequestClaimed(address indexed user, uint256 indexed id, Fair amount, Timestamp claimDate);
+
+    error NoStakeToClaim();
+
+    // internal
+
+    function createRequest(ExitQueue storage queue, address user, Fair amount) internal {
+        
+        UserExitQueue storage userQueue = queue.userQueues[user];
+        Fair oldAmount = Fair.wrap(0);
+        if (userQueue.firstId != 0 && userQueue.lastId + 1 - userQueue.firstId == queue.maxRequests) {
```

Comment:
You are overriding the newest one.
If you override the oldest one you can use `_claim` and not duplicate the code.

Reply:
I also recommend to emit an event logging the fact that the request was overwritten.